### PR TITLE
[FIN] Insetting APIs for Rect

### PR DESCRIPTION
**Pull request by saniul**
Implements https://github.com/Khan/Prototope/issues/50

- [x] Swift API
- [x] JS Bridge
- [x] Documentation

### CODE REVIEW

File: PrototopeJSBridge/GeometryBridge.swift [hunk 1]
```diff
@@ -173,6 +174,31 @@ import Prototope
...
+        }
+        let getValue = { (key: String) -> Double in
+            return (args.objectForKey(key) as! Double?) ?? 0
+        }
+        
+        let justValue = hasKey("value")
+        let simplified = hasKey("vertical") || hasKey("horizontal")
+        let core = hasKey("top") || hasKey("right") || hasKey("bottom") || hasKey("left")
+        
+        switch (justValue, simplified, core) {
```
Comment (jbrennan) [inline PrototopeJSBridge/GeometryBridge.swift]: What do you think about using ifs here instead? This way, you can avoid the truth-table switch statement, which might make the intention a little clearer:

```swift
if let justValue = hasKey("value") {
    return RectBridge(rect.inset(value: getValue("value")))
}

if let simplified = hasKey("vertical") || hasKey("horizontal") {
    return ...
}
// ...

Environment.currentEnvironment?.exceptionHandler("Trying to call inset on with invalid parameters: \(args)")
return self
```

File: Examples/TestInsets/main.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* 
+
+And example to demonstrate how to use insets on Rects
+
+*/
+
+
+var r1 = new Rect({x:100, y:100, width:250, height:250})
+
+console.log(r1.minX+","+r1.minY+":"+r1.maxX+","+r1.maxY);
```
Comment (andymatuschak) [inline Examples/TestInsets/main.js]: Ooh, that reminds me: we should override `toString()` on these things. Filed #59.

File: PrototopeJSBridge/GeometryBridge.swift (hunk 1 above)
Comment (andymatuschak) [inline PrototopeJSBridge/GeometryBridge.swift]: Think this is a good suggestion. Reading the `case`s, I find myself having to look up at the switch expression to remind myself which is which.

**Comment by andymatuschak** [comment]
Very sorry for the super long turnaround on this. API surface looks good!

**Comment by saniul** [comment]
Don't apologise, I myself haven't really had time to work on any of the outstanding PRs this past week :(

File: PrototopeJSBridge/GeometryBridge.swift (hunk 1 above)
Comment (saniul) [inline PrototopeJSBridge/GeometryBridge.swift]: Ah, the point here was to make sure that users of this API don’t try to mix and match the possible sets arguments.

Comment (andymatuschak) [inline PrototopeJSBridge/GeometryBridge.swift]: Ah, I see. I think that's worth doing: otherwise, they could end up with silent failures. :+1:

**Comment by saniul** [comment]
I guess that’s it for this PR. gh-pages docs will come separately

**Comment by andymatuschak** [comment]
woooo!

**Pull request merged**